ceph-dev-build/build/validate_rpm

```diff
@@ -11,3 +11,7 @@ set -ex
       exit 1
       ;;
   esac) || exit 0
+
+if [ "${DIST}" == "windows" ]; then
+  exit 0
```

Comment:
I think it's supposed to return 0 for the targeted job type and 1 otherwise. Wondering if we should return 1 and maybe move this check a bit higher.

Reply:
That `if` statement is never going to run because the one above it will either `exit 1` or `0` before it gets a chance.

May I suggest

1. Create a function in `scripts/build_utils.sh`.  Maybe call it `is_windows()` and put that `if` statement in it to return `true` or `false`.  Or whatever a bashier way is.
2. Run the `is_windows` function before the `exit` in each Ubuntu and Windows case.  (e.g., [here](https://github.com/ceph/ceph-build/blob/master/ceph-dev-build/build/validate_deb#L7))

Reply:
Example `validate_deb`

```
#!/bin/bash
set -ex

# Only do actual work when we are a DEB distro
( source /etc/os-release
  case $ID in
    debian|ubuntu)
      if is_windows; then
        exit 1
      fi
      exit 0
      ;;
    *)
      exit 1
      ;;
  esac) || exit 0
  ```

Then

```
is_windows() {
  if [ "${DIST}" == "windows" ]; then
    return
  fi
  false
}
```

Reply:
Actually, the script will only exit if that `case` statement exits w/ return code `1`. The code is very confusing at initial glance but it's relying on the fact that the `case` statement within the parathesis is run as a separate script, so the `exit` calls only affect that sub-script. Then if it exits with 0 (i.e. continue), we don't hit the `||` case and continue. If it exits with 1 (i.e. stop the build), it hits the `||` and exits the build cleanly.

Reply:
> Create a function in scripts/build_utils.sh. Maybe call it is_windows() and put that if statement in it to return true or false. Or whatever a bashier way is.

Right now the `validate_XYZ` scripts are invoked before the `build_utils.sh` are pulled in. I figured a repeated three line `if` statement was better than totally restructuring at this point.

Reply:
Fair enough.  Sounds like we need to rethink the validation scripts anyway